setup/includes/upgrades/common/3.0.2-remove-deprecated-resource-fields.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/**
+ * Scripts removes deprecated fields from the main resource table
+ *
+ * @var modX $modx
+ * @package setup
+ */
+
+use MODX\Revolution\modResource;
+
+$class = modResource::class;
+$table = $modx->getTableName($class);
+
+$deprecatedFields = [
+    'privateweb',
+    'privatemgr',
+    'donthit',
+];
+
+foreach ($deprecatedFields as $deprecatedField) {
+    $description = $this->install->lexicon('drop_column', ['column' => $deprecatedField, 'table' => $table]);
+    $this->processResults($class, $description, [$modx->manager, 'removeField'], [$class, $deprecatedField]);
+}
```

Comment:
I think this should be refactored to drop all the columns in a single statement. You'll have to use the `ALTER TABLE DROP COLUMN column1, column2, column3` syntax, but this will be much more efficient, especially on sites with a large number of resources.

Reply:
Fair enough, I’ll try to find some time to do so next week.

Reply:
@opengeek @Mark-H done